lib/AppInfo/Application.php

```diff
@@ -26,12 +26,33 @@
 
 namespace OCA\DroneciFastLane\AppInfo;
 
+use OCA\DroneciFastLane\Listener\TalkListener;
+use OCA\Talk\Chat\ChatManager;
+use OCP\App\IAppManager;
 use OCP\AppFramework\App;
+use OCP\AppFramework\Bootstrap\IBootContext;
+use OCP\AppFramework\Bootstrap\IBootstrap;
+use OCP\AppFramework\Bootstrap\IRegistrationContext;
+use OCP\EventDispatcher\IEventDispatcher;
 
-class Application extends App {
+class Application extends App implements IBootstrap {
 	public const APP_ID = 'droneci_fast_lane';
 
 	public function __construct() {
 		parent::__construct(self::APP_ID);
 	}
+
+	public function register(IRegistrationContext $context): void {
+	}
+
+	public function boot(IBootContext $context): void {
+		/** @var IAppManager $appManager */
+		$appManager = $context->getServerContainer()->get(IAppManager::class);
+
+		if ($appManager->isEnabledForUser('spreed')) {
+			/** @var IEventDispatcher $dispatcher */
+			$dispatcher = $context->getServerContainer()->get(IEventDispatcher::class);
+			$dispatcher->addListener(ChatManager::EVENT_BEFORE_MESSAGE_SEND, [TalkListener::class, 'handleCommand']);
```

Comment:
These events are not meant to be "public" until we switched to the new and "final" typed events.

That being said, any reason you don't want to use the command things we provide? I guess because a /help would then list the command although it doesn't work everywhere?
Something to keep in mind for the next iteration of Talk commands I guess.

Reply:
> That being said, any reason you don't want to use the command things we provide?

I actually did look into that, but there were a few obstacles that made me end up following this approach. From what I remember one was that commands can be registered via occ only, there was no PHP API. I looked for register methods, but could not find any. Also I recalled only CLI scripts were supported, but with a fresh look this does not seem to be the case anymore if it used to be like this at all.

My preference would actually be to use what is provided by Talk. For now I also like to have this working on 24 as well.

Let me look into this again… when i use private Talk components anyway, at least I can try to only utilize the `CommandService`.

Reply:
Totally fine for me, fixing in case we rename or something can be easily done, so fine by me

Reply:
For the record, as discussed with @nickvergessen in chat, we stay with this approach.